compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -497,15 +498,15 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     /// `with_caller_location` indicates whether the caller passed a caller location. Miri
     /// implements caller locations without argument passing, but to match `FnAbi` we need to know
     /// when those arguments are present.
-    pub(crate) fn eval_fn_call(
+    pub fn eval_fn_call(
```

Comment:
I'm not happy with making this more public, this is a rather low-level function. Can this be avoided?

Reply:
It can probably be avoided with the `return Result<Option<Instance>>` scheme.

Reply:
I feel like returning `Option<Instance>` is the same level of awkwardness...

Reply:
This PR has the awkwardness of attempting to split "find the fn" from "call the fn", but then in quite a few cases "call the fn" ends up finding MIR elsewhere, completely breaking the structure.

If it is okay for `call_extra_fn` to call `eval_fn_call`, then we're just back to where we were before but with more messy code, I think? I don't see how you gained anything.